VibeSomeMauiBro/Services/CatService.cs

```diff
@@ -0,0 +1,107 @@
+using System.Text.Json;
+using VibeSomeMauiBro.Models;
+
+namespace VibeSomeMauiBro.Services;
+
+public interface ICatService
+{
+    Task<List<Cat>> GetCatsAsync(int count = 10);
+    Task<List<Cat>> GetLikedCatsAsync();
+    Task LikeCatAsync(Cat cat);
+    Task DislikeCatAsync(Cat cat);
+}
+
+public class CatService : ICatService
+{
+    private readonly HttpClient _httpClient;
+    private readonly List<Cat> _likedCats = new();
```

Comment:
Use a `HashSet<Cat>` for `_likedCats` to improve performance.

Then you don't need the Contains() check below either.